### PR TITLE
Amp-compare-slider: Brand New Component Skeleton

### PR DESCRIPTION
**Pull request by alannawalton**
All the files associated with this component. Implementation is in a different PR .

### CODE REVIEW

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import {Layout} from '../../../src/layout';
+
+export class AmpCompareSlider extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.myText_ = 'hello world';
```
Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: remove this. Add it back if used later

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /** @private {!Element} */
+    this.container_ = this.win.document.createElement('div');
```
Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: Please delay document API until `#buildCallback`. Only initiate `this.container_` in `#constructor`, and assign its value in `#buildCallback`

File: extensions/amp-compare-slider/amp-compare-slider.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# <a name="`amp-compare-slider`"></a> `amp-compare-slider`
+
+<table>
+  <tr>
+    <td width="40%"><strong>Description</strong></td>
+    <td>FILL THIS IN</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Availability</strong></td>
+    <td>FILL THIS IN</td>
```
Comment (zhouyx) [inline extensions/amp-compare-slider/amp-compare-slider.md]: Indicate that the component is under development.

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  /** @override */
+  buildCallback() {
+    this.container_.textContent = this.myText_;
+    this.element.appendChild(this.container_);
+    this.applyFillContent(this.container_, /* replacedContent */ true);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.RESPONSIVE;
```
Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: What's our conclusion here? Do we only support layout responsive?  to @camelburrito @ericfs

Comment (ericfs) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: Was this actually intended for me? I don't think I'm familiar with the context here.

Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: ooops sorry, meant to cc @ericlindley-g  on the layout format😄

Comment (camelburrito) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: i think we will need to support everything except container .

Comment (ericlindley-g) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: :)   I think we'd want to support all standard layout types (fill, fixed, fixed-height, flex-item, nodisplay, responsive), unless there's a reason not to. WDYT?

/cc @aghassemi  for visibility

**Comment by alannawalton** [comment]
Can I get another look at this pull request please?

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js (hunk 4 above)
Comment (aghassemi) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: chatted with Alanna offline, yes all but container

**Comment by zhouyx** [comment]
@alannawalton Please fix the travis build failure. Try `gulp lint` locally.

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    super(element);
+
+    /** @private {?Element} */
+    this.container_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.container_ = this.win.document.createElement('div');
+    this.container_.textContent = this.myText_;
```
Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: `this.myText_` is undefined. Please remove this line.

File: extensions/amp-compare-slider/0.1/amp-compare-slider.js (hunk 4 above)
Comment (zhouyx) [inline extensions/amp-compare-slider/0.1/amp-compare-slider.js]: @alannawalton Let's add a TODO here. And add support to other layout in following PR.

**Comment by zhouyx** [comment]
@alannawalton Almost there. But travis is failing, please fix the test.

**Comment by zhouyx** [comment]
🎉 🎉 🎉

**Pull request merged**